File: dp/level2/LC_Jump Game II.py
import sys

class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)
        dp = [sys.maxsize] * n
        dp[n - 1] = 0
        for i in range(n - 2, -1, -1):
            for j in range(i + 1, i + nums[i] + 1):
                if j >= n:
                    break
                dp[i] = min(dp[i], dp[j] + 1)
        return dp[0]

File: dp/level2/LC_Jump_Game_II.java
package dp.level2;
import java.util.*;

public class LC_Jump_Game_II {
	public int jump(int[] nums) {
		int[] dp = new int[nums.length];
		Arrays.fill(dp, Integer.MAX_VALUE);
		dp[0] = 0;
		for (int i = 0; i < nums.length; i++) {
			int step = nums[i];
			for (int s = 0; s <= step; s++) {
				if (i + s < nums.length) {
					dp[i + s] = Math.min(dp[i + s], dp[i] + 1);
				}
			}
		}
		return dp[nums.length - 1];
	}
}
